File: go-handler-plugin/core/receiver.go
package receiver

import (
	"github.com/extism/go-pdk"
)

// Execute the function
func SetHandler(function func(param []byte) ([]byte, error))  {

	functionParameters := pdk.Input()

	value, err := function(functionParameters)

	stringValue := string(value)

	var stringError string
	if err != nil {
		stringError = err.Error()
	} else {
		stringError = ""
	}

	jsonResult := `{"success":"`+stringValue+`", failure:"`+stringError+`"}`

	mem := pdk.AllocateString(jsonResult)
	// copy output to host memory
	pdk.OutputMemory(mem)
	//return 0

}

File: js-handler-plugin/src/core/receiver.js
function setHandler(func) {
	// read function argument from the memory
	let input = Host.inputString()

	let res = func(input)

	// copy output to host memory
	Host.outputString(JSON.stringify({
		success: res[0], 
		failure: res[1]
	}))

	return 0
}

module.exports = {setHandler}

